samples/xamarin/ToDoItems/ToDoItems.Core/Services/CosmosDBService.cs

```diff
@@ -0,0 +1,136 @@
+﻿using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+
+using Microsoft.Azure.Documents;
+using Microsoft.Azure.Documents.Client;
+using System;
+using System.Data;
+using System.Diagnostics;
+using Microsoft.Azure.Documents.Linq;
+using Xamarin.Forms;
+
+namespace ToDoItems.Core
+{
+    public class CosmosDBService
+    {
+        static DocumentClient docClient = null;
+
+        static readonly string databaseName = "Tasks";
+        static readonly string collectionName = "Items";
+
+        static async Task<bool> Initialize()
+        {
+            if (docClient != null)
+                return true;
+
+            try
+            {
+                docClient = new DocumentClient(new Uri(APIKeys.CosmosEndpointUrl), APIKeys.CosmosAuthKey);
```

Comment:
Please error out in-case the defaults are not updated by customer. This caused CRI's in the past.

Reply:
@kirankumarkolli - [this](https://github.com/codemillmatt/azure-documentdb-dotnet/blob/5d942e6d9fa6f6b0158d4df715053ebfb8bf6c99/samples/xamarin/ToDoItems/ToDoItems.Core/Helpers/APIKeys.cs#L10) will stop the user from compiling until they entered both the auth key & url.

Reply:
+1. I like it.